Crash in Card Browser, the multi-select mode: Edit => change the card type

###### Reproduction Steps

1. Create any type and clone it (to keep things simple, I've just used basic, calling the types "Basic 1" & "Basic 2").
2. Add two cards to the first type.
3. Browser → select the two new cards → Edit → try to change the type (in my case, "Basic 1" → "Basic 2"). 


###### Expected Result
If you could allow there some sort of batch type reassignment via the default field mapping, that would be perfect!


###### Actual Result

It crashes, only the first card is affected by the change.


###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

###### Research
*Enter an [x] character to confirm the points below:*

--- I've reported it via the play.google.com, your representative there suggested that I report it here.


- [ ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [ ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

> I think #6243 might need a revert now that I think about it - there may be other note-related things that would confuse people if they thought it would apply to all the selected notes but it really just affects one, and the feature of changing note type for multiple notes at once is still on the wish list

It was #6243. Apologies.

To note: #6243 was intended to only show the "edit note" dialog if all selected cards are on the same note. It should not have appeared when multiple notes of different types were selected.

This is an issue between changing the note type while we are in multiselect mode.

<details><summary>Stack Trace</summary>


```
2020-06-11 14:11:59.122 24055-24149/com.ichi2.anki E/com.ichi2.anki: [qarth_debug:]  get PatchStore::createDisableExceptionQarthFile method fail.
2020-06-11 14:11:59.140 24055-24149/com.ichi2.anki E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #4
    Process: com.ichi2.anki, PID: 24055
    java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$3.done(AsyncTask.java:355)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:246)
        at java.util.concurrent.ThreadPoolExecutor.processTask(ThreadPoolExecutor.java:1187)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:784)
     Caused by: java.lang.IndexOutOfBoundsException: Index: 1, Size: 0
        at java.util.ArrayList.get(ArrayList.java:437)
        at com.ichi2.async.CollectionTask.doInBackgroundCheckCardSelection(CollectionTask.java:1573)
        at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:355)
        at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:73)
        at android.os.AsyncTask$2.call(AsyncTask.java:334)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:246) 
        at java.util.concurrent.ThreadPoolExecutor.processTask(ThreadPoolExecutor.java:1187) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:784) 
```
</details>

Data flow:

* `onActivityResult` -> save card edit -> search cards

----

* `onCreateOptionsMenu` -> `mInMultiSelectMode = true`
* `updateMultiselectMenu`
* `TASK_TYPE_CHECK_CARD_SELECTION` -> crash. This task is passed in both the card positions and the card lists. These are out of sync due to the searchCards call.

--- a/AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java
@@ -657,6 +657,8 @@ public class CardBrowser extends NavigationDrawerActivity implements
         editCard.putExtra(NoteEditor.EXTRA_CALLER, NoteEditor.CALLER_CARDBROWSER_EDIT);
         editCard.putExtra(NoteEditor.EXTRA_CARD_ID, sCardBrowserCard.getId());
         startActivityForResultWithAnimation(editCard, EDIT_CARD, ActivityTransitionAnimation.LEFT);
+        //#6432 - FIXME - onCreateOptionsMenu crashes if receiving an activity result from edit card when in multiselect
+        endMultiSelectMode();
     }
 
     private void openNoteEditorForCurrentlySelectedNote() {

--- a/AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java
@@ -125,7 +125,7 @@ public class CardBrowser extends NavigationDrawerActivity implements
     public static final String EDITED = "edited";
     public static final String INTERVAL = "interval";
     public static final String LAPSES = "lapses";
-    public static final String NOTE = "note";
+    public static final String NOTE_TYPE = "note";
     public static final String REVIEWS = "reviews";
     private static Pattern sMarkedPattern = Pattern.compile(".*[Mm]arked.*");
 
@@ -180,7 +180,7 @@ public class CardBrowser extends NavigationDrawerActivity implements
     private static final String[] COLUMN2_KEYS = {ANSWER,
         CARD,
         DECK,
-        NOTE,
+        NOTE_TYPE,
         QUESTION,
         TAGS,
         LAPSES,
@@ -1512,7 +1512,7 @@ public class CardBrowser extends NavigationDrawerActivity implements
                 break;
         }
         item.put(LAPSES, Integer.toString(c.getLapses()));
-        item.put(NOTE, c.model().optString("name"));
+        item.put(NOTE_TYPE, c.model().optString("name"));
         item.put(QUESTION, formatQA(q, context));
         item.put(REVIEWS, Integer.toString(c.getReps()));
         String tags = note.stringTags();

--- a/AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java
@@ -380,30 +380,9 @@ public class CardBrowser extends NavigationDrawerActivity implements
         return ids;
     }
 
-    private boolean hasSelectedSingleNoteId() {
-        //Heuristic to skip a large array copy
-        if (checkedCardCount() > 50) {
-            return false;
-        }
-        //copy to array to ensure threadsafe iteration
-        Integer[] checkedPositions = mCheckedCardPositions.toArray(new Integer[0]);
-        List<Map<String, String>> cards = mCards;
-
-        HashSet<String> notes = new HashSet<>();
-        for (Integer position : checkedPositions) {
-            String noteId;
-            try {
-                noteId = cards.get(position).get(NOTE);
-            } catch (IndexOutOfBoundsException e) {
-                //#6384
-                Timber.w(e, "concurrent modification of mCards array - assume more than one note selected");
-                return false;
-            }
-            if (notes.add(noteId) && notes.size() > 1) {
-                return false;
-            }
-        }
-        return mCards == cards && notes.size() == 1;
+    private boolean canPerformMultiSelectEditNote() {
+        //The noteId is not currently available. Only allow if a single card is selected for now.
+        return checkedCardCount() == 1;
     }
 
     @VisibleForTesting
@@ -884,7 +863,7 @@ public class CardBrowser extends NavigationDrawerActivity implements
         mActionBarMenu.findItem(R.id.action_select_all).setVisible(!hasSelectedAllCards());
         //Note: Theoretically should not happen, as this should kick us back to the menu
         mActionBarMenu.findItem(R.id.action_select_none).setVisible(hasSelectedCards());
-        mActionBarMenu.findItem(R.id.action_edit_note).setVisible(hasSelectedSingleNoteId());
+        mActionBarMenu.findItem(R.id.action_edit_note).setVisible(canPerformMultiSelectEditNote());
     }
 
 

